src/main/java/com/amazonaws/services/kinesis/clientlibrary/config/DatePropertyValueDecorder.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Amazon Software License (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/asl/
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.services.kinesis.clientlibrary.config;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Provide Date property.
+ */
+class DatePropertyValueDecoder implements IPropertyValueDecoder<Date> {
+
+    /**
+     * Constructor.
+     */
+    DatePropertyValueDecoder() {
+    }
+
+    /**
+     * @param value property value as String
+     * @return corresponding variable in correct type
+     */
+    @Override
+    public Date decodeValue(String value) {
+        try {
+            return new Date(Long.parseLong(value) * 1000L);
```

Comment:
Just calling this out for myself, and any future readers: 

If someone accidentally uses milliseconds this won't work the way they expect.  In the future it might make sense to use Instant from java.time.  Instant#parse can deal with textual date representations, and can be converted to a java.util.Date pretty easily.